src/java.base/share/classes/com/sun/crypto/provider/AESCipher.java

```diff
@@ -138,19 +138,19 @@ public AES256_CFB_NoPadding() {
             super(32, "CFB", "NOPADDING");
         }
     }
-    public static final class AES128_GCM_NoPadding extends OidImpl {
+    public static final class AES128_GCM_NoPadding extends GaloisCounterMode {
         public AES128_GCM_NoPadding() {
-            super(16, "GCM", "NOPADDING");
+            super(16, new AESCrypt());
         }
     }
-    public static final class AES192_GCM_NoPadding extends OidImpl {
+    public static final class AES192_GCM_NoPadding extends GaloisCounterMode {
         public AES192_GCM_NoPadding() {
-            super(24, "GCM", "NOPADDING");
+            super(24, new AESCrypt());
         }
     }
-    public static final class AES256_GCM_NoPadding extends OidImpl {
+    public static final class AES256_GCM_NoPadding extends GaloisCounterMode {
         public AES256_GCM_NoPadding() {
-            super(32, "GCM", "NOPADDING");
+            super(32, new AESCrypt());
         }
     }
```

Comment:
These should be removed since SunJCE registers com.sun.crypto.provider.GaloisCounterMode$AES128/AES192/AES256 instead of these?

Reply:
In addition, the private boolean field "updateCalled" of this class and its relevant assignments can also be removed.

Reply:
Good catch on some of these older files that I changed.. I hadn't even thought about going back to check on them